Coral/interactive.js

```diff
@@ -52,18 +61,16 @@ export class InteractiveTangent {
     const uv = this.tile_quad.world_to_uv(mouse);
     const original_tangent = uv.sub(this.control_point.position);
 
-    // If the tangent is facing backwards, negate it to flip it forwards.
-    // This is done instead of a reflection so we don't get sudden jumps
-    // when the mouse crosses the 90 degree line.
-    const flipped_tangent =
-      original_tangent.dot(this.constraint) < 0
-        ? original_tangent.neg()
+    // If the tangent is facing backwards, reflect it so it's facing forwards
+    const corrected_tangent =
+      original_tangent.dot(this.forward_direction) < 0
+        ? this.flip_forward.transform(original_tangent).neg()
```

Comment:
Add a note here explaining that `neg()` is needed here because this is a bivector, not a vector.